File: app/src/main/java/com/nuhkoca/offlineapp/binding/DateParsingBindingAdapter.java
package com.nuhkoca.offlineapp.binding;

import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

import androidx.databinding.BindingAdapter;

public class DateParsingBindingAdapter {

    public DateParsingBindingAdapter() {
    }

    @BindingAdapter(value = "android:date")
    public void bindDate(TextView target, Date date) {
        SimpleDateFormat formatter = new SimpleDateFormat("MMM dd", new Locale("tr", "TR", "tr"));
        target.setText(formatter.format(date));
    }
}

File: app/src/main/java/com/nuhkoca/offlineapp/binding/DateParsingBindingAdapter.kt
package com.nuhkoca.offlineapp.binding

import android.widget.TextView
import androidx.databinding.BindingAdapter
import java.text.SimpleDateFormat
import java.util.*

class DateParsingBindingAdapter {

    @BindingAdapter(value = ["android:date"])
    fun bindDate(target: TextView, date: Date) {
        val formatter = SimpleDateFormat("MMM dd", Locale("tr", "TR", "tr"))
        target.text = formatter.format(date)
    }
}
